splitter.go

```diff
@@ -38,25 +41,19 @@ func (r SplitReader) ReadTag(name string, value interface{}) error {
 			if !ok {
 				continue // Can't touch that
 			}
-			fieldName = name + "." + fieldName // add prefix
+			if name != "" {
+				fieldName = name + "." + fieldName // add prefix
+			}
 			field := str.Field(i)
-			if !field.CanAddr() {
-				err = fmt.Errorf("Cannot address %s", fieldName)
+			err = r.readValue(fieldName, field)
```

Comment:
How do we feel about the `if err = r.readValue(fieldName, field); err != nil` idiom?

Reply:
I'm inconsistent about using it.

But in this case I think there's good reason to avoid it (maybe?).

That idiom is usually used when the returned variable is only locally scoped (e.g. when the error is immediately returned, or a value is used in the block). So the idiom is usually `:=`. 

In this case the `err` is used later, so while the idiom would still work, I think it might deceive the reader. The `err` was used later because at one point I had some code that executed after the switch. (Or at least intended to move in that direction.)

... however, now that the code is effectively `break; return err` in those cases, I think the idiom would be more readable here. So good point: I'll refactor this a bit to return instead of break, using the idiom.